x/metadata/keeper/msg_server.go

```diff
@@ -48,6 +59,44 @@ func (k msgServer) WriteScope(
 	return types.NewMsgWriteScopeResponse(msg.Scope.ScopeId), nil
 }
 
+func (k msgServer) createExpirationForScope(ctx sdk.Context, msg *types.MsgWriteScopeRequest) (*exptypes.Expiration, error) {
+	// create expire scope request.
+	expireMsg := types.NewMsgExpireScopeRequest(msg.Scope.ScopeId, msg.Signers)
+	wrapper, anyErr := codec.NewAnyWithValue(expireMsg)
+	if anyErr != nil {
+		return nil, anyErr
+	}
+
+	// use default expiration, but override with cmd line arg if provided
+	expShort := k.GetDefaultScopeExpiration(ctx, msg.Scope)
+	if len(msg.Expiration) > 0 {
+		expShort = msg.Expiration
+	}
```

Comment:
No need to make this keeper call if we aren't using the default.

```suggestion
	expShort := msg.Expiration
	if len(expShort) == 0 {
		expShort = k.GetDefaultScopeExpiration(ctx, msg.Scope)
	}
```

Reply:
updated as suggested.